src/server/pps/server/api_server.go

```diff
@@ -1092,17 +1016,18 @@ func convertDatumState(state datum.State) pps.DatumState {
 
 func (a *apiServer) collectDatums(ctx context.Context, job *pps.Job, cb func(*datum.Meta, *pfs.File) error) error {
 	jobInfo, err := a.InspectJob(ctx, &pps.InspectJobRequest{
-		Job: client.NewJob(job.ID),
+		Job: job,
 	})
 	if err != nil {
 		return err
 	}
 	pachClient := a.env.GetPachClient(ctx)
-	fsi := datum.NewCommitIterator(pachClient, jobInfo.StatsCommit)
+	metaCommit := ppsutil.MetaCommit(jobInfo.OutputCommit)
```

Comment:
Probably should handle a service pipeline job the same way as the PR referenced above. So, just return an error that the job does not have datums associated with it (even though it does, this probably needs more thought later).

Reply:
So this will just error out based on what was discussed in the thread above, which should be fine.